src/plugins/intel_gpu/src/kernel_selector/cl_kernels/grid_sample_ref.cl

```diff
@@ -1,40 +1,240 @@
-// Copyright (C) 2022 Intel Corporation
-// SPDX-License-Identifier: Apache-2.0
-//
-
-KERNEL(grid_sample_ref)
-(const __global INPUT0_TYPE* data, const __global INPUT1_TYPE* grid, __global OUTPUT_TYPE* output) {
-    const uint bf = get_global_id(0);
-    const uint b = bf % OUTPUT_BATCH_NUM;
-    const uint f = bf / OUTPUT_BATCH_NUM;
-    const uint y = get_global_id(1);
-    const uint x = get_global_id(2);
-
-#ifdef ALIGN_CORNERS
-    printf("ALIGN_CORNERS = true\n");
+typedef __global INPUT0_TYPE data_t;
+typedef __global INPUT1_TYPE grid_t;
+typedef __global OUTPUT_TYPE output_t;
+
+typedef INPUT0_TYPE data_et;
+typedef INPUT1_TYPE grid_et;
+typedef OUTPUT_TYPE output_et;
+
+inline const data_et FUNC(
+    get_data_single_value)(const data_t* buffer, const size_t n, const size_t c, const size_t h, const size_t w) {
```

Comment:
@ozhydkov-lohika I'm not sure that we need to support so big "size_t" type. In most cases, we use "unit" for indexes.